sphinxcontrib/dotnetdomain.py

```diff
@@ -256,16 +255,38 @@ def run(self):
         return nodes
 
     def before_content(self):
-        '''Build up prefix with nested elements'''
+        '''Build up prefix history for nested elements
+
+        The following keys are used in :py:attr:`self.env.temp_data`:
+
+            dn:prefixes
+                Stores the prefix history. With each nested element, we add the
+                prefix to a list of prefixes. When we exit that object's nesting
+                level, :py:meth:`after_content` is triggered and the prefix is
+                removed from the end of the list.
+
+            dn:prefix
+                Current prefix. This should reflect the last element in the
+                prefix history
+        '''
         super(DotNetObjectNested, self).before_content()
         if self.names:
             (parent, prefix) = self.names.pop()
-            self.env.temp_data['dn:prefix'] = prefix
-            self.prefix_set = True
+            try:
+                self.env.temp_data['dn:prefixes'].append(prefix)
```

Comment:
It would clean it up a little, though `env.temp_data` is created on the parent `ObjectDescription` or above on `Directive`, I'd be worried about side effects of changing that here.